### ISSUE TITLE
Clearer explanation of validation set

### ISSUE THREAD

**Issue by dafnevk** [carpentries-incubator/deep-learning-intro#211]
(copied from #207)
In Episode 3, the "Watch your model training closely" section describes using a validation set to avoid "iterating through countless cycles of model trainings and subsequent evaluations with a reserved test set". That gives the impression that the validation set is being used for convenience. It seems important to stress (in the episode or instructor notes) that using the test set repeated to assess model or hyperparameter performance is an invalid workflow because the model will be indirectly fit to the test set. Then the test set can no longer be used to estimate generalization performance on new data, which is usually the real goal.
